### ISSUE TITLE
The suitability of cherry_crispr

### ISSUE THREAD

**Issue by danzheng7372** [KennthShang/CHERRY_crispr_DATABASE#2]
Hi, 
Thank you for developing this cherry_crispr tool for host prediction. My question is about the suitability of the tool. Does cherry_crispr function for integrated viral sequence identified from MAGs (e.g. prokaryotic MAGs assembled from pellet microbial community samples) or free viral MAGs (e.g. viral MAGs assembled from supernatant viral community samples)?

Best,
Dan

**Comment by KennthShang** [comment]
I am sorry that I can not fully catch up with your questions (since I am not a 100% biologist). Maybe you can describe your question in a more understandable way.

Currently, the two extensions of CHERRY (CHERRY_crispr_database: this folder and CHERRY_crispr_MAGs: https://github.com/KennthShang/CHERRY_crispr_MAG) supply two different ways for users who only want to use CRISPR for host prediction.

CHERRY_crispr_database: ONLY used the CRISPRs found in GenBank database
CHERRY_crispr_MAGs: ALLOW users to use their own prokaryotic MAGs for prediction

I think your question may refer to the latter one?

Best,
Jiayu

**Comment by danzheng7372** [comment]
> I am sorry that I can not fully catch up with your questions (since I am not a 100% biologist). Maybe you can describe your question in a more understandable way.
> 
> Currently, the two extensions of CHERRY (CHERRY_crispr_database: this folder and CHERRY_crispr_MAGs: https://github.com/KennthShang/CHERRY_crispr_MAG) supply two different ways for users who only want to use CRISPR for host prediction.
> 
> CHERRY_crispr_database: ONLY used the CRISPRs found in GenBank database CHERRY_crispr_MAGs: ALLOW users to use their own prokaryotic MAGs for prediction
> 
> I think your question may refer to the latter one?
> 
> Best, Jiayu

Hi Jiayu,
Thank you very much for the quick reply. I am sorry that I didn't describe my question well. When CHERRY_crispr_MAGs is used for host prediction, the input ".fa" file should be the viral sequence, in my case I have two different types of viral contigs: 1) viral contigs identified from prokaryotic MAGs using tools like "virsorter", "vibrant" or "phabox (phamer)"; 2) viral genomes (free viral MAGs) assembled from virus sequencing data (The virus DNA is extracted from supernatant viral community samples). So does CHERRY_crispr_MAGs function for these two viral contigs?

Best regards,
Dan

**Comment by KennthShang** [comment]
Of course, the program can run on both types of data. The logic of the program is: 1. Find the CRISPRs from prokaryotes. 2. Run BLASTN using your viral FASTA as a query against the CRISPR database.

However, if you are using viral MAGs, you may need to be careful about the accession/ID of your contigs within MAGs. The alignment program will only be conducted on the sequences/contigs level rather than automatically concatenating all your contigs in the viral MAGs. Thus, there are two options for your reference:

1. Write a for loop to run your MAGs iteratively, and it will return results for each MAG independently.
2. Renaming the accession/ID in your viral MAGs and combine all the MAGs into one FASTA. After running the CHERRY_crispr, you can map the accession/ID to your original MAGs.


Best,
Jiayu

**Comment by danzheng7372** [comment]
Hi Jiayu,
Thank you for the reply. And I use for loop to run all viral MAGs, that works well.

Best regards,
Dan

**Comment by KennthShang** [comment]
Good to know!

Best,
Jiayu
